types/src/main/scala/aqua/types/Type.scala

```diff
@@ -398,6 +398,8 @@ case class StreamMapType(override val element: DataType) extends MutableStreamTy
     StructType(name, NonEmptyMap.of("key" -> ScalarType.string, "value" -> element))
 
   def toCanon: ImmutableCollectionType = CanonStreamMapType(element)
+
+  def toProduct: ProductType = ProductType(ScalarType.string :: element :: Nil)
```

Comment:
nit: I would call it `elementProduct`, bc it is unclear on the first glance how to transform map type to product